### PR TITLE
update partition

### PR DESCRIPTION
**Pull request by daviszhen**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #8312 
issue #8315 
issue #6591 
issue #7717 

## What this PR does / why we need it:

1, adjust fields in the `mo_tables`
Add field `partition_info` holds the information about partitions in the table.
`partition_info`,BLOB, saves serialized partition information.

`partitioned`,int, denotes the table be split or not.

2, add partition table holds the data for the partition.
When a table is partitioned, its data composed by several pieces from partition tables.
The DDL of the table should be changed.

3, adjust the semantic verification for the partition.
Update the KEY partition and LIST partition(partial)

4, add some fields in the plan.proto

### CODE REVIEW

File: pkg/vm/engine/disttae/types.go [hunk 1]
```diff
@@ -192,7 +192,8 @@ type txnTable struct {
 	clusterByIdx int // -1 means no clusterBy key
 	viewdef      string
 	comment      string
-	partition    string
+	partitioned  int8   //1 : the table has partitions ; 0 : no partition
```
Comment (zhangxu19830126) [inline pkg/vm/engine/disttae/types.go:195]: why not bool?

**Review by zhangxu19830126 — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

File: proto/plan.proto [hunk 2]
```diff
@@ -337,6 +337,7 @@ message PartitionByDef {
 	int64 algorithm = 7;
 	bool is_sub_partition = 8;
 	string partition_msg = 9;
+	repeated string partition_table_names = 10;
```
[the remark this answers is not in the document]
Comment (daviszhen) [reply proto/plan.proto:340]: 理论上可以。
0.8 暂时做不到。truncate 之后tableid 会变，所有子表的tableID都会变。就需要把partitionByDef取出来，改变tableId再填进去。这就麻烦了。
0.9 等tableid不变后，可以再改。

**Review by nnsgmsone — COMMENTED** [review]
Can this be done in plan?

**Review by nnsgmsone — APPROVED** [review]

**Comment by daviszhen** [comment]
@mergifyio refresh

**Comment by daviszhen** [comment]
@mergifyio refresh

**Pull request merged**